### PR TITLE
Disallow fields in service constructor

### LINKED ISSUE #17304: Disable Fields in Service constructor
**Description:**
Spec grammar 
```
service-constructor-expr := [annots] service service-body-block
service-body-block := { service-method-defn* }
service-method-defn :=
   metadata
   service-method-defn-quals
   function identifier function-signature method-body
service-method-defn-quals :=
   [explicit-visibility-qual | resource]
```

### LINKED ISSUE #17465: Tooling updates needed for field variable related changes in services
PR #17408 removes support for field variables in service constructor. The tools need to reflect this change as well.

### PR DESCRIPTION
**Pull request by pubudu91**
Fixes #17304
Fixes #17465 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-integration-test/src/test/resources/http/src/httpservices/29_listener_action_service.bal [hunk 1]
```diff
@@ -18,21 +18,21 @@ import ballerina/http;
 
 listener http:Listener echoEP  = new(9229);
 
+string serviceLevelStringVar = "sample value";
```
Comment (MaryamZi) [inline tests/jballerina-integration-test/src/test/resources/http/src/httpservices/29_listener_action_service.bal]: Shall we rename this variable?

File: tests/jballerina-unit-test/src/test/resources/test-src/statements/variabledef/globalcycle/src/viaservice/file01.bal [hunk 2]
```diff
@@ -22,19 +22,19 @@ string b7a = "b7a";
 int port = o.p;
 
 Obj o = new();
+string str = "";
+int p = port;
 
 service hello on lis {
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/resources/test-src/statements/variabledef/globalcycle/src/viaservice/file01.bal]: Do we need this service at all for this test?

File: tests/jballerina-integration-test/src/test/resources/grpc/src/grpcservices/03_bidirectional_streaming_service.bal [hunk 3]
```diff
@@ -21,6 +21,9 @@ import ballerina/runtime;
 // Server endpoint configuration
 listener grpc:Listener ep3 = new (9093);
 
+map<grpc:Caller> consMap = {};
```
Comment (MaryamZi) [inline tests/jballerina-integration-test/src/test/resources/grpc/src/grpcservices/03_bidirectional_streaming_service.bal]: Should we verify this change with the standard library team? I guess the examples would also have to change, maybe we can do that when revisiting the BBEs.

Can we also check if we can rename this variable? `consMap` sounds a bit like a const map.

File: tests/jballerina-unit-test/src/test/resources/test-src/statements/variabledef/globalcycle/src/viaservice/file01.bal (hunk 2 above)
Comment (pubudu91) [reply tests/jballerina-unit-test/src/test/resources/test-src/statements/variabledef/globalcycle/src/viaservice/file01.bal]: Not sure. The `__init()` of the service contains an assignment using global vars. Shall we just leave it as is at least for now? The checks passed previously

File: tests/jballerina-integration-test/src/test/resources/grpc/src/grpcservices/03_bidirectional_streaming_service.bal (hunk 3 above)
Comment (daneshk) [reply tests/jballerina-integration-test/src/test/resources/grpc/src/grpcservices/03_bidirectional_streaming_service.bal]: Yes, we have a similar sample in examples as well. will check it later.
This is a connection map. It's ok to change it to any meaningful name.

File: compiler/ballerina-lang/src/main/resources/grammar/BallerinaParser.g4 [hunk 4]
```diff
@@ -44,12 +44,7 @@ serviceDefinition
     ;
 
 serviceBody
-    :   LEFT_BRACE serviceBodyMember* RIGHT_BRACE
-    ;
-
-serviceBodyMember
-    :   objectFieldDefinition
-    |   objectFunctionDefinition
+    :   LEFT_BRACE objectFunctionDefinition* RIGHT_BRACE
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/resources/grammar/BallerinaParser.g4]: We seem to have another deviation. Resource functions cannot have visibility qualifiers. Let's fix that separately as discussed offline. Created https://github.com/ballerina-platform/ballerina-lang/issues/17607 to track that.

File: tests/jballerina-integration-test/src/test/resources/grpc/src/grpcservices/03_bidirectional_streaming_service.bal [hunk 5]
```diff
@@ -21,6 +21,9 @@ import ballerina/runtime;
 // Server endpoint configuration
 listener grpc:Listener ep3 = new (9093);
 
+map<grpc:Caller> constMap = {};
```
Comment (MaryamZi) [inline tests/jballerina-integration-test/src/test/resources/grpc/src/grpcservices/03_bidirectional_streaming_service.bal]:
```suggestion
map<grpc:Caller> connMap = {};
```

File: tests/jballerina-unit-test/src/test/resources/test-src/statements/variabledef/globalcycle/src/viaservice/file01.bal (hunk 2 above)
Comment (hasithaa) [reply tests/jballerina-unit-test/src/test/resources/test-src/statements/variabledef/globalcycle/src/viaservice/file01.bal]: Please refer https://github.com/ballerina-platform/ballerina-spec/issues/289

**Pull request merged**